torch/csrc/generic/StorageMethods.cpp

```diff
@@ -108,16 +102,22 @@ static PyObject * THPStorage_(resize_)(PyObject *_self, PyObject *number_arg)
   THPUtils_assert(THPUtils_checkLong(number_arg), "resize_ expects an int, "
       "but got %s", THPUtils_typename(number_arg));
   int64_t newsize = THPUtils_unpackLong(number_arg);
-#if defined(THC_GENERIC_FILE)
-  ptrdiff_t size_bytes_i = newsize;
-  TORCH_CHECK(!c10::overflows<size_t>(size_bytes_i),
-              "Requested storage size (", size_bytes_i,
-              ") cannot be represented as a size_t");
-  const auto size_bytes = static_cast<size_t>(size_bytes_i);
-  at::native::resize_bytes_cuda(self->cdata, size_bytes);
-#else
-  at::native::resize_bytes_cpu(self->cdata, newsize);
+  c10::DeviceType device_type = self->cdata->device_type();
+  if (device_type == at::kCPU) {
+    at::native::resize_bytes_cpu(self->cdata, newsize);
+#ifdef USE_CUDA
+  } else if (device_type == at::kCUDA) {
+    ptrdiff_t size_bytes_i = newsize;
+    TORCH_CHECK(!c10::overflows<size_t>(size_bytes_i),
+                "Requested storage size (", size_bytes_i,
+                ") cannot be represented as a size_t");
+    const auto size_bytes = static_cast<size_t>(size_bytes_i);
+    at::native::resize_bytes_cuda(self->cdata, size_bytes);
 #endif
+  } else {
+    TORCH_CHECK(false,
+      "_UntypedStorage.resize_: got unexpected device type ", device_type);
```

Comment:
Ditto here, the data pointer should know how to resize itself, so this should be possible to make generic (but as before, this could be done in follow up)